src/http/server.ts

```diff
@@ -328,13 +340,17 @@ export class Server {
           status: response.status,
         });
       } catch (e) {
-        if (isNaN(e.code)) {
-          e.code = 500;
+        if (errorHandler) {
+          try {
+            await errorHandler.catch(e, originalRequest, response);
+          } catch (e) {
+            await defaultErrorHandler.catch(e, originalRequest, response);
+          }
+        } else {
+          await defaultErrorHandler.catch(e, originalRequest, response);
         }
-        return new Response(e.stack, {
-          status: e.code,
-          headers: response.headers,
-        });
+
+        return new Response(response.body, response);
```

Comment:
```ts
try {
  // stuff
} catch (e) {
  await this.#error_handler.catch(e, originalRequest, response);
}
```

Reply:
we use defaultErrorHandler when `this.#error_handler.catch()` or `errorHandler.catch()` (it's the same `const errorHandler = this.#error_handler;`) thrown an error